baseline.yaml

```diff
@@ -196,18 +196,24 @@ criteria:
     maturity_level: 1
     category: Build & Release
     criteria: |
-      The project's build and release pipelines
-      MUST NOT execute arbitrary code that is
-      input from outside of the build script.
+      Build and release pipelines MUST NOT execute
+      untrusted code.
     objective: |
       Reduce the risk of code injection or other
       security vulnerabilities in the project's
       build and release processes by restricting
-      the execution of external code.
+      the execution of external code in workflows.
     implementation: |
       Ensure that the project's build and release
-      pipelines do not execute arbitrary code
-      provided from external sources.
+      pipelines do not execute untrusted code
+      provided from external sources. Maintainers
+      may establish trust in a variety of ways,
+      including digital signature verification and
+      inspection of external code. For clarity,
+      this criterion does not prohibit the use of
+      software in a package format that executes
+      scripts (e.g. RPM, .deb) so long as the
+      package itself is trusted.
```

Comment:
I'm struggling with how to enforce this.  For example, does this prohibit using a tool built and maintained in the same repository which automates some of the release steps unless we can ensure that the tool cannot be subverted?  As written, it suggests every Makefile and data processor needs a security review, which seems high for maturity level 1.

Similarly, it's not clear whether this criteria establishes a transitive duty of care to audit not just your own codebase, but those of pipeline dependencies (e.g., do I need to check on untrusted code execution from `actions/setup-go` in my GitHub workflows)?  Or can I say "I'm using this action from $ELSEWHERE, I assume it's secure" at level 1, and have additional security criteria at level 2 or 3?

Reply:
It does not establish a transitive duty of care. As an example, a workflow that uses an Ubuntu image from Canonical wouldn't violate this because it's a trusted source. A workflow that uses `some-random-rebuild/ubuntu-latest` would, unless the project has established trust in the `some-random-rebuild` project. The idea is more along the lines of "we're getting the code we think we're getting" as opposed to "we have inspected every line of software all the way down and it is clean".

For the first part, I would say if it's in the same repository, it's part of the project and therefore not an "external source".

Reply:
> As an example, a workflow that uses an Ubuntu image from Canonical wouldn't violate this because it's a trusted source.

But it _would_ violate this requirement as written. The term "trusted source" isn't present in this text.

I'll try to make a change, I think the point is that build only runs code in the repo *or* from external trusted sources.